### PR TITLE
Add user client to allow for suspending/unsuspending users

### PR DESCRIPTION
**Pull request by dennisgranath**
What:
Adds a user client to allow for suspending/unsuspending users.

### CODE REVIEW

File: src/main/java/com/spotify/github/v3/clients/UserClient.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+   *
+   * @param username username of the user to suspend
+   * @return a CompletableFuture that indicates success or failure
+   */
+  public CompletableFuture<Integer> suspendUser(
+      final String username, final SuspensionReason reason) {
+    final String path = String.format(SUSPEND_USER_TEMPLATE, username);
+    return github
+        .put(path, github.json().toJsonUnchecked(reason))
+        .thenApply(resp -> resp.code() == NO_CONTENT ? 1 : 0);
```
Comment (felix-seifert) [inline src/main/java/com/spotify/github/v3/clients/UserClient.java:52]: Is `0` and `1` a suitable solution here? I don't know the exact requirements for this method but I feel that a `boolean` would be nicer.

File: src/main/java/com/spotify/github/v3/clients/UserClient.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+   *
+   * @param username username of the user to unsuspend
+   * @return a CompletableFuture that indicates success or failure
+   */
+  public CompletableFuture<Integer> unSuspendUser(
+      final String username, final SuspensionReason reason) {
+    final String path = String.format(SUSPEND_USER_TEMPLATE, username);
+    return github
+        .delete(path, github.json().toJsonUnchecked(reason))
+        .thenApply(resp -> resp.code() == NO_CONTENT ? 1 : 0);
```
Comment (felix-seifert) [inline src/main/java/com/spotify/github/v3/clients/UserClient.java:66]: Same as above

**Review by felix-seifert — COMMENTED** [review]
Are these change ideas applicable?

File: src/main/java/com/spotify/github/v3/clients/UserClient.java (hunk 1 above)
Comment (dennisgranath) [reply src/main/java/com/spotify/github/v3/clients/UserClient.java:52]: Sure! Let's go with boolean.

**Review by felix-seifert — APPROVED** [review]
:rocket:

**Pull request merged**